hibernate-core/src/main/java/org/hibernate/engine/internal/StatefulPersistenceContext.java

```diff
@@ -684,6 +684,7 @@ public Object narrowProxy(Object proxy, EntityPersister persister, EntityKey key
 
 			// Otherwise, create the narrowed proxy
 			final HibernateProxy narrowedProxy = (HibernateProxy) persister.createProxy( key.getIdentifier(), session );
```

Comment:
This is not a safe thing to do because there can be other references to the original proxy which might cause problems when the session is flushed. Also, the original proxy will still have a reference to the `Session` but the `Session` knows nothing about the original proxy any more. When the `Session` is closed, proxies stored in the `PersistenceContext` are detached by nullifying the `Session` reference. Since the original proxy is not longer in the `Session` it will continue to have a reference to the now closed `Session`.